<launch>
    <arg name="model_name" default="albatross"/>
    <arg name="config_pkg" default="$(eval find(arg('model_name') + '_description'))"/>

    <arg name="controller_configs" default="$(arg config_pkg)/controllers.yaml"/>

    <group ns="$(arg model_name)/controllers">
        <rosparam file="$(arg controller_configs)" command="load"/>

        <node name="cascaded_controller" pkg="tauv_common" type="cascaded_controller" output="screen"/>

        <group ns="cascaded_controller/pids">
            <node name="pos_roll_controller" pkg="pid" type="controller"/>
            <node name="pos_pitch_controller" pkg="pid" type="controller"/>
            <node name="pos_yaw_controller" pkg="pid" type="controller"/>
            <node name="pos_x_controller" pkg="pid" type="controller"/>
            <node name="pos_y_controller" pkg="pid" type="controller"/>
            <node name="pos_z_controller" pkg="pid" type="controller"/>
            <node name="vel_roll_controller" pkg="pid" type="controller"/>
            <node name="vel_pitch_controller" pkg="pid" type="controller"/>
            <node name="vel_yaw_controller" pkg="pid" type="controller"/>
            <node name="vel_x_controller" pkg="pid" type="controller"/>
            <node name="vel_y_controller" pkg="pid" type="controller"/>
            <node name="vel_z_controller" pkg="pid" type="controller"/>
        </group>
    </group>

</launch>